src/t_stream.c

```diff
@@ -985,8 +985,16 @@ static int streamParseAddOrTrimArgsOrReply(client *c, streamAddTrimArgs *args, i
 
             if (args->approx_trim) {
                 /* In order to prevent from trimming to do too much work and cause
-                 * latency spikes we limit the amount of work it can do */
-                args->limit = 100 * server.stream_node_max_entries; /* Maximum 100 rax nodes. */
+                 * latency spikes we limit the amount of work it can do.
+                 * Maybe we want to limit the listpack only by `stream_node_max_bytes`, 
+                 * thus setting `stream_node_max_entries` to zero, but this will result 
+                 * in unlimited trim. 
+                 * In the case of larger stream_node_max_entries, it is actually 
+                 * meaningless, so we limit the effective range to [1,1000000].
+                 */
+                if (args->limit < 1) args->limit = 1;
+                if (args->limit > 1000000) args->limit = 1000000;
+                args->limit *= 100;
```

Comment:
but now there's no reference to `server.stream_node_max_entries`
i meant something like 
```
args->limit = 100 * server.stream_node_max_entries; /* Maximum 100 rax nodes. */
if (args->limit < 1) args->limit = 1;
if (args->limit > 1000000) args->limit = 1000000;
```

another small comment about the C comment you wrote: i feel like it's a bit too long, maybe we can shorten it to
```
We have to cap args->limit from both sides in case stream_node_max_entries is 0 or too big (could cause overflow)
```

Reply:
I seem to have done something stupid just now, please treat it as if you didn’t see it, thank you very much! :)